### ISSUE TITLE
[Process Part] Get Proposals For Call

### ISSUE THREAD

**Issue by florian-papsdorf** [hpi-swt2/contacts-portal#129]
# Description
Als Benutzer möchte ich auf der Startseite Vorschläge für spontane digitale Treffen mit verfügbaren Kontakten bekommen.

# Dependencies
- [ ] meinem Konto können Kontakte durch Kontaktanfragen zugeordnet werden

# Conditions of satisfaction
- [ ] es gibt einen Button, um die Vorschläge zu aktualisieren
- [ ] auf jeder Karte ist das Profilfoto sowie der vollständige Name zu sehen
- [ ] auf jeder Karte ist ein Button, um einen Call per Jitsi zu starten
- [ ] auf jeder Karte sind bis zu 3 weitere Handles, über die ich die Kontakte erreichen kann
- [ ] auf jeder Karte ist eine Anzeige, wann ich den Vorschlag erhalten habe
- [ ] wenn ich auf das Call-Icon drücke, Wechsel ich in ein Jitsi-Call mit dem User
- [ ] ich bekomme nur Vorschläge, wenn ich den Status "Nice2MeetU" oder "Available" gesetzt habe
- [ ] ich bekomme maximal 4 Vorschläge
- [ ] vorgeschlagene Kontakte haben auch den Status "Nice2MeetU" oder "Available" gesetzt
- [ ] Vorschläge basieren sowohl auf Leuten mit denen ich sehr häufig, sehr wenig Kontakt habe oder erst neulich als Kontakt gesetzt habe

# Zugehöriges Wireframe

![Change Status zu NiceToMeetU, Get Proposal](https://user-images.githubusercontent.com/49521681/104849618-033fbe00-58eb-11eb-86b8-35f0f3300c27.jpg)

**Comment by inadicis** [comment]
Zwei Anmerkungen hier:
>  auf jeder Karte ist eine Anzeige, wann ich den Vorschlag erhalten habe

Das hätte ich nicht so erwartet, sondern dass jedes mal, das der Nutzer auf der Seite geht, wird ihn 3 (aktuelle) Vorschlage generiert. Aus alte Vorschläge zeigen würde das Problem folgen, dass diese Leute vielleicht nicht mehr online / im nice2MeetU  status

>  Vorschläge basieren sowohl auf Leuten mit denen ich sehr häufig, sehr wenig Kontakt habe oder erst neulich als Kontakt gesetzt habe

Das klingt implizit nach ein kluges matching-Algorithmus. Ich finde es nicht realistisch das an den Kunde zu verkaufen. Ich würde zu erst ein ganz dummes gleichverteiltes Random match als erste Lösung erwarten.

**Comment by florian-papsdorf** [comment]
Genau, immer, wenn ich Refresh klicke, werden neue Vorschläge generiert. Wie wäre es, wenn die Zeit darstellt, wann die entsprechende Person ihren Status auf Nice2MeetU oder Available geändert hat?


Ja, wir müssen uns zuerst mit einem "kleineren" Vorschlagssystem auseinandersetzen, aber ich habe das in den Prozess genommen, weil es ja langfristig einen Mehrwert bietet - vlt. ist das genau ein Punkt, den wir mit dem Kunden abklären ;)

**Comment by inadicis** [comment]
> 
> 
> Genau, immer, wenn ich Refresh klicke, werden neue Vorschläge generiert. Wie wäre es, wenn die Zeit darstellt, wann die entsprechende Person ihren Status auf Nice2MeetU oder Available geändert hat?

Ach so hatte ich es nicht verstanden. Ich finde es eine gute Idee!

> 
> Ja, wir müssen uns zuerst mit einem "kleineren" Vorschlagssystem auseinandersetzen, aber ich habe das in den Prozess genommen, weil es ja langfristig einen Mehrwert bietet - vlt. ist das genau ein Punkt, den wir mit dem Kunden abklären ;)

Fair enough.
